src/client/dfs/SConscript

```diff
@@ -7,49 +7,24 @@ def scons():
 
     denv = env.Clone()
 
-    """If Lustre installed use its headers instead of our local defines"""
-    try:
-        with open('/etc/os-release') as f:
-            for line in f:
-                if 'ID' in line:
-                    line = line.strip()
-                    _, distro = line.strip().split('=')
-                    distro = distro.strip('"')
-                    break
-        if distro in ['centos', 'rhel', 'fedora', 'opensuse']:
-            import rpm
-
-            gotit = False
-            ts = rpm.TransactionSet()
-            headers = ts.dbMatch('name', 'lustre')
-            if not headers:
-                headers = ts.dbMatch('name', 'lustre-client')
-            if headers:
-                from pkg_resources import parse_version
-
-                for header in headers:
-                    if parse_version(header['version']) >= parse_version('2.12.57'):
-                        denv.AppendUnique(CCFLAGS=['-DLUSTRE_INCLUDE'])
-                        gotit = True
-            if gotit == False:
-                print("Found no compatible lustre nor lustre-client package.")
-        elif distro in ('debian', 'ubuntu'):
-            import apt
-            from pkg_resources import parse_version
-
-            cache = apt.Cache()
-            cache.open()
-            instver = cache["lustre"].installed
-            if not instver:
-                instver = cache["lustre-client"].installed
-            if instver and parse_version(instver.version) >= parse_version('2.12.57'):
-                denv.AppendUnique(CCFLAGS=['-DLUSTRE_INCLUDE'])
-            else:
-                print("Found no compatible lustre nor lustre-client package.")
+    """ If Lustre installed build a Lustre-aware libduns """
+    conf = Configure(denv)
+    gotversion = False
+    if not conf.CheckLibWithHeader('lustreapi', 'linux/lustre/lustre_user.h', 'c'):
+        print("No installed Lustre version detected")
+    else:
+        print("Installed Lustre version detected")
+        if not conf.CheckFunc('llapi_unlink_foreign'):
+            print("Lustre version is not compatible")
         else:
-            print("Unable to identify distro.")
-    except ImportError:
-        pass
+            print("Lustre version is compatible")
+            gotversion = True
+    if gotversion is True:
+        print("Building with Lustre bindings.")
+        conf.env.AppendUnique(CCFLAGS=['-DLUSTRE_INCLUDE'])
+    else:
+        print("Not building with Lustre bindings.")
+    denv = conf.Finish()
```

Comment:
ah, I see.   I've never seen this before.

Reply:
I found this behaviour documented at "https://github.com/SCons/scons/wiki/SconsAutoconf"